buildkitd/buildkitd.go

```diff
@@ -186,6 +186,11 @@ func RemoveExited(ctx context.Context) error {
 
 // Start starts the buildkitd daemon.
 func Start(ctx context.Context, image string, settings Settings, reset bool) error {
+	err := CheckCompatibility(ctx, settings)
+	if len(settings.AdditionalArgs) == 0 && err != nil {
```

Comment:
Assume that providing buildkit args means you are aware you have compatibility issues and go ahead anyways.